whynotearth/BrowTricks
# Feature: Line Item UI Component

# Description

This does not address a specific issue, but rather creates a component for the common UI element seen here in the Figma designs:

![image](https://user-images.githubusercontent.com/20192983/81261809-918c4200-8ff1-11ea-9221-21a5a2070217.png)

## Implementation

We make a "LineItem" component which has 

1. 40x40 image
2. three lines of text (call them "title", "subtitle", and "description", where "description" is optional)
3. optional right-aligned element (call it "detail") which could have a click action

We use slots to make each part (image, "title", "subtitle", "description", and "detail") configurable with their own text and styles. "Detail" can even be an SVG component.

See:

* [src/components/ui/LineItem.vue](https://github.com/whynotearth/BrowTricks/pull/44/files#diff-9f6e499ccfb0d7cdb103078d18542477R1-R29) for the component code

and

* [src/views/TestLineItemComponent.vue](https://github.com/whynotearth/BrowTricks/pull/44/files#diff-e90b909f57fa96c502b12bc30722fbdfR1-R127) for example instantiation code (MenuHeader/MenuDrawer, etc. content here other than LineItem is irrelevant for the review).

# How to test

Go to [/test-line-item-component](https://deploy-preview-44--browtricks.netlify.app/test-line-item-component) (link will take you to the deploy preview) and observe three examples of the "LineItem" component usage.

Note that clicking the "detail" on the second "LineItem" component will trigger an alert, which is just an example of us putting a click event on such a "detail".
